packages/material-ui/src/Snackbar/Snackbar.js

```diff
@@ -95,181 +94,157 @@ export const styles = theme => {
   };
 };
 
-class Snackbar extends React.Component {
-  state = {};
+const Snackbar = React.forwardRef(function Snackbar(props, ref) {
+  const {
+    action,
+    anchorOrigin: { vertical, horizontal },
+    autoHideDuration,
+    children,
+    classes,
+    className,
+    ClickAwayListenerProps,
+    ContentProps,
+    disableWindowBlurListener,
+    message,
+    onClose,
+    onEnter,
+    onEntered,
+    onEntering,
+    onExit,
+    onExited,
+    onExiting,
+    onMouseEnter,
+    onMouseLeave,
+    open,
+    resumeHideDuration,
+    TransitionComponent,
+    transitionDuration,
+    TransitionProps,
+    ...other
+  } = props;
 
-  componentDidMount() {
-    if (this.props.open) {
-      this.setAutoHideTimer();
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.open !== this.props.open) {
-      if (this.props.open) {
-        this.setAutoHideTimer();
-      } else {
-        clearTimeout(this.timerAutoHide);
-      }
-    }
-  }
-
-  componentWillUnmount() {
-    clearTimeout(this.timerAutoHide);
-  }
-
-  static getDerivedStateFromProps(nextProps, prevState) {
-    if (typeof prevState.exited === 'undefined') {
-      return {
-        exited: !nextProps.open,
-      };
-    }
-
-    if (nextProps.open) {
-      return {
-        exited: false,
-      };
-    }
-
-    return null;
-  }
+  const timerAutoHide = React.useRef();
+  const [exited, setExited] = React.useState();
```

Comment:
Looking at this again you need to set a default value for edited

Reply:
Done, and just like that all my tests pass 🎸 🎉 Can you kindly review 😄